### ISSUE TITLE
New Remap function for parse_syslog_facility

### ISSUE THREAD

**Issue by timansky** [vectordotdev/vrl#848]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Similar to request to map syslog severity levels to integer from https://github.com/vectordotdev/vector/issues/5501 it would be helpful to have a means to remap [syslog facility keywords](https://en.wikipedia.org/wiki/Syslog#Facility) to their integer values as well.

### Use Cases

When parsing events message, the raw facility values could be enriched with a simple remap transform to down stream processing

### Proposal

create a remap-function to_syslog_facility which could produce the facility name:

parse_syslog_facility("authpriv")
=> 10

**Comment by jszwedko** [comment]
Makes sense to me, thanks for the request @timansky . We'd be happy to see a PR for this if you or someone else feels motivated.

**Comment by timansky** [comment]
I'll try, looks simple

**Comment by jszwedko** [comment]
> I'll try, looks simple

🥳 You can see the PR you linked to, https://github.com/vectordotdev/vector/issues/5501, as an example.
